src/main/java/ro/msg/learning/shop/service/OrderService.java

```diff
@@ -0,0 +1,75 @@
+package ro.msg.learning.shop.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import ro.msg.learning.shop.dto.CreateOrderDTO;
+import ro.msg.learning.shop.dto.CreateOrderDetailDTO;
+import ro.msg.learning.shop.exception.OnlineShopException;
+import ro.msg.learning.shop.mapper.AddressMapper;
+import ro.msg.learning.shop.model.Order;
+import ro.msg.learning.shop.model.OrderDetail;
+import ro.msg.learning.shop.model.Stock;
+import ro.msg.learning.shop.repository.LocationRepository;
+import ro.msg.learning.shop.repository.OrderRepository;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class OrderService {
+
+    private final OrderRepository orderRepository;
+    private final LocationRepository locationRepository;
+    private final AddressMapper addressMapper;
+
+    @Transactional
+    public Order createOrder(CreateOrderDTO createOrderDTO) {
+        final var productIdToQuantity = createOrderDTO.orderDetails().stream()
+                .collect(Collectors.toMap(CreateOrderDetailDTO::productId, CreateOrderDetailDTO::quantity));
+        final var location = locationRepository.findAll().stream()
```

Comment:
Can you write a query for this complex operation? It is very expensive to retrieve all the locations first and then filter the wrong ones out. If you have hundreds of locations and only one of these locations is the correct one, you will load all the locations in memory which is an expensive operation.